packages/anoncreds/src/protocols/proofs/v1/V1ProofProtocol.ts

```diff
@@ -64,20 +54,17 @@ import {
 import { V1PresentationProblemReportMessage } from './messages/V1PresentationProblemReportMessage'
 import { V1PresentationPreview } from './models/V1PresentationPreview'
 
-type IndyProofFormatServiceLike = ProofFormatService<IndyProofFormat>
-
 export interface V1ProofProtocolConfig {
-  // indyCredentialFormat must be a service that implements the `IndyProofFormat` interface, however it doesn't
-  // have to be the IndyProofFormatService implementation per se.
-  indyProofFormat: ProofFormatService<IndyProofFormat>
+  indyProofFormat: LegacyIndyProofFormatService
 }
 
-export class V1ProofProtocol extends BaseProofProtocol implements ProofProtocol<[IndyProofFormatServiceLike]> {
-  private indyProofFormat: ProofFormatService<IndyProofFormat>
+export class V1ProofProtocol extends BaseProofProtocol implements ProofProtocol<[LegacyIndyProofFormatService]> {
+  private indyProofFormat: LegacyIndyProofFormatService
 
   public constructor({ indyProofFormat }: V1ProofProtocolConfig) {
     super()
 
+    // TODO: just create a new instance of LegacyIndyProofFormatService here so it makes the setup easier
```

Comment:
This is true, as we are always using a `LegacyIndyProofFormatService`. It will represent that we'll use separate instances for V1 and V2 protocols. If we foresee that most agents will support both protocols, it won't be so different in terms of simplicity.

Reply:
My initial approach was that you register credential formats also in the agent and the protocols could hook into it. But I changed it so you have to provide the formats yourself. Do you think we should change it?

What do you think will be the simplest way to give users flexibility?

Reply:
Oh, I think I misunderstood the comment. I thought that you would like to remove indyProofFormat from constructor parameters and simply do a `this.indyProofFormat = new LegacyIndyProofFormatService()`.

But I think it's fine the way it is, letting to provide the formats we want.